share/man/man9/cdefs.9

```diff
@@ -0,0 +1,297 @@
+many environments
+.Fx
+supports compilation for.
+It defines convenience macros for the
+.Fx
+sources, tailored to the base
+system's needs.
+.Pp
+Most of these macros are for use inside the
+.Fx
+sources only.
+They are not intended as a general portability layer.
+.Sh Supported Compilers
+There are two levels of compiler support for
+.Fx :
+Compilers supported for building programs on
+.Fx .
+These are compilers that you can use to build and run programs on
+Fx .
+.Bl -column -offset indent ".Sy Compiler" ".Sy Versions"
+.It Sy Compiler Ta Sy Versions
+.It gcc Ta 9, 10, 11, 12, 13, 14
+.It clang Ta 10, 11, 12, 13, 14, 15, 16, 17, 18
+.It TinyC (tcc) Ta 0.9
+.It pcc Ta 1.1
+.El
+Due to testing constraints, tcc and pcc may not always work.
+.PP
+Compilers supported for building
+.Fx
+itself:
+.Bl -column -offset indent ".Sy Compiler" ".Sy Versions"
+.It Sy Compiler Ta Sy Versions
+.It gcc Ta 12, 13
+.It clang Ta 16, 17, 19
+.El
+.Sh Macros and Magic for Programming Environment
+.Nm
+defines (or refrains from defining) a number of macros to increase portability
+of compiled programs.
+These are to allow more advanced language features to appear in header files.
+The header files assume a compiler that accepts C prototype function
+declarations.
+They also assume bare ansi keywords can be used.
+The
+.Dv __P
+macro is deprecated, will be removed in the future and not even documented.
+.Ss General Macros
+.Bl -column ".Sy Macro" ".Sy Description"
+.It Sy Macro Ta Sy Description
+.It Dv __const Ta expands to const in ansi and newer environments,
+__const in pre-ansi environments that support this extension or nothing
+otherwise.
+This my be removed soon, since it is unused in our tree.
+.It Dv __signed Ta expands to signed in ansi and newer environments,
+__signed in pre-ansi environments that support this extension or nothing
+otherwise.
+This my be removed soon, since it is unused in our tree.
+.It Dv __volatile Ta expands to volatile in ansi and newer environments,
+__volatile in pre-ansi environments that support this extension or nothing
+otherwise.
+.It Dv __inline Ta expands to inline in C++ and ansi and newer environments,
+__inline in pre-ansi environments that support this extension or nothing
+otherwise.
+.It Dv __restrict Ta expands to restrict in C99 and newer environments, to nothing otherwise.
+.It Dv __CONCAT Ta used to paste two pre-processor tokens.
+.It Dv __STRING Ta used to convert the argument to a string.
+.It Dv __BEGIN_DECLS Ta Start a group of functions.
+.It Dv __END_DECLS Ta End a group of functions.
+In a C environment, these are defined as nothing.  In a C++ environment, these
+declare the functions to have
+.Dq C
+linkage.
+.El
+.Pp
+When
+.Dv NO_ANSI_KEYWORDS
+is defined in the compilation environment,
+const, inline, signed and volatile are defined to be nothing in pre-ansi
+compilation environments.
+NOTE: this may disappear soon too, since it's been broken for years.
```

Comment:
```suggestion
This has been broken for years, it may disappear soon.
```

Reply:
IT turns out, it's disappearing now, so I deleted all this text.